### ISSUE TITLE
unable to open some psd files

### ISSUE THREAD

**Issue by mylastgg** [psd-tools/psd-tools#97]
Hello!
I'm using the latest version of psd_tools to batch-process some psd files.
The process breaks for some specific psd files, apparently random:

 File “/Users/davide/Desktop/cruncher-internal/src/cruncher.py”, line 48, in exportExerciseJson
   psd = PSDImage.open(filename)
 File “/usr/local/lib/python3.7/site-packages/psd_tools/api/psd_image.py”, line 101, in open
   self = cls(PSD.read(f))
 File “/usr/local/lib/python3.7/site-packages/psd_tools/psd/__init__.py”, line 72, in read
   LayerAndMaskInformation.read(fp, encoding, header.version),
 File “/usr/local/lib/python3.7/site-packages/psd_tools/psd/layer_and_mask.py”, line 57, in read
   return cls._read_body(f, encoding, version)
 File “/usr/local/lib/python3.7/site-packages/psd_tools/psd/layer_and_mask.py”, line 64, in _read_body
   global_layer_mask_info = GlobalLayerMaskInfo.read(fp)
 File “/usr/local/lib/python3.7/site-packages/psd_tools/psd/layer_and_mask.py”, line 914, in read
   data = read_length_block(fp)
 File “/usr/local/lib/python3.7/site-packages/psd_tools/utils.py”, line 77, in read_length_block
   length = read_fmt(fmt, fp)[0]
 File “/usr/local/lib/python3.7/site-packages/psd_tools/utils.py”, line 36, in read_fmt
   len(data), fmt_size
AssertionError: read=3, expected=4

This is the error log

Maybe I'm missing something?

**Comment by kyamagu** [comment]
@mylastgg Thanks for reporting a but. Can you share any sample file?

**Comment by mylastgg** [comment]
Sorry I am not sure I can distribute the files for copyright reasons. I'll ask my colleagues. 
In the meantime, I've noticed that this only happens for some files after I've resized them using imagemagick like this:

`'convert ' + filename + ' -matte +distort SRT "0,0  .25  0  0,0" ' + filename`

so maybe I am not using imagemagick correctly to resize the psd.
Side question: is psd_tools able to resize a psd file?
